mysys/my_thr_init.c

```diff
@@ -354,7 +354,7 @@ void my_thread_end()
       tmp->dbug= NULL;
     }
     free(tmp);
-
+    set_mysys_thread_var(NULL);
```

Comment:
Please, also add a comment here regarding why `set_mysys_thread_var(NULL)` should be called before signalling. The description form the commit message is precise enough.